presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -252,6 +252,9 @@ default ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTabl
         throw new PrestoException(NOT_SUPPORTED, "This connector does not support deletes");
     }
 
+    default void beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
It is optional because it is used while planning, so the layout might not be known until later in the planning stage. However, the layout must definitely exist by the time the query starts executing, since the layout is used to generate splits in `ConnectorSplitManager`.